src/pykeen/models/predict.py

```diff
@@ -115,6 +114,143 @@ def _get_input_batch(
     return target, batch, (batch_ids[0], batch_ids[1])
 
 
+def _get_mapped_triples(mapped_triples: Union[CoreTriplesFactory, MappedTriples]):
+    """Get mapped triples."""
+    if isinstance(mapped_triples, CoreTriplesFactory):
+        mapped_triples = mapped_triples.mapped_triples
+    return mapped_triples
+
+
+class PredictionPostProcessor:
+    """A post-processor for predictions."""
+
+    def __init__(self, **filter_triples: Union[None, CoreTriplesFactory, MappedTriples]) -> None:
+        """Instantiate the processor.
+
+        :param filter_triples:
+            the filter triples
+        """
+        self.filter_triples = {
+            key: _get_mapped_triples(value) for key, value in filter_triples.items() if value is not None
+        }
+
+    @abstractmethod
+    def _contains(self, df: pd.DataFrame, mapped_triples: MappedTriples, invert: bool = False) -> numpy.ndarray:
+        """
+        Return which of the rows of the given data frame are contained in the ID-based triples.
+
+        :param df: nrows: n
+            the predictions
+        :param mapped_triples: shape: (m, 3)
+            the ID-based triples
+        :param invert:
+            whether to invert the result
+
+        :return: shape: (n,), dtype: bool
+            a boolean mask indicating which row is contained in the given ID-based triples
+        """
+        raise NotImplementedError
+
+    def filter(self, df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Filter out known triples.
```

Comment:
should mention in docstring this is not an in-place operation, maybe should give the possibility for both ways?